### PR TITLE
Migrate tests from MaxIT to MaxAggregatorTests

### PR DESCRIPTION
**Pull request by csoulios**
This PR migrates tests from MaxIT integration test to MaxAggregatorTests, as described in #42893

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java [hunk 1]
```diff
@@ -24,338 +24,24 @@
...
-import static org.elasticsearch.search.aggregations.AggregationBuilders.terms;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchResponse;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.notNullValue;
 
-public class MaxIT extends AbstractNumericTestCase {
+public class MaxIT extends ESIntegTestCase {
+
```
Comment (jimczi) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java]: I think it's ok to remove this test entirely. `testDontCacheScripts` is tested on a lot of aggregations (metrics, top_hits, terms, ...) but we don't really need an integration test for this. For instance `AbstractQueryTestCase#testToQuery` checks `QueryShardContext#isCacheable` after building the query, we could have the same kind of unit test in `MaxAggregatorTests`. Does this makes sense ?

**Review by jimczi — COMMENTED** [review]
The move of the tests look good to me but I left one comment regarding the remaining integration test.

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java (hunk 1 above)
Comment (polyfractal) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java]: ++ Christos and I chatted about this last week and agreed it's doable in the framework.  I'll make the change and merge (Christos was on support ramp and wasn't able to get to it before PTO).  Thanks!

Comment (csoulios) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java]: Removed integration test `MaxIT` and implemented caching checks in `MaxAggregatorTests#testCacheAggregation` (without scripts -> cached) and `MaxAggregatorTests#testDontCacheScripts` (with scripts -> not cached)

**Review by jimczi — APPROVED** [review]
LGTM

**Review by polyfractal — APPROVED** [review]
Also LGTM! :)

**Pull request merged**